airflow/providers/google/cloud/hooks/dataprep.py

```diff
@@ -34,11 +34,23 @@ class GoogleDataprepHook:
     """
 
     def __init__(self) -> None:
-        self.headers = {
+        self.headers = None
+        self._url = URL
+
+    @property
+    def headers(self):
+        """
+        To limit logic in __init__ method request headers are moved to setter.
+        """
+        return self._headers
+
+    @headers.setter
+    def headers(self, value):  # pylint: disable=unused-argument
```

Comment:
Here is a `value` agr but it's unused and I don't get why setter needs such arg -- and it works now.

Reply:
Why you need setter?

Reply:
Anyway, setter needs this argument because it needs a `value` to set it for a corresponding property. Simples example:

```python
@property
def magic_value(self):
    return self._value

@magic_value.setter
def magic_value(self, value):
    self._value = value
```

If you will do 
```
@magic_value.setter
def magic_value(self, value):
    self. magic_value = value
```
then it may result in recursion error (setter calls itself)

Reply:
Thanks for explanation.